src/hooks/useFetchUrlContent.js

```diff
@@ -1,59 +1,73 @@
-  const [isLoading, setIsLoading] = useState(false);
-  const [error, setError] = useState(null);
-
-  const checkChromeStorageUsage = useCallback(async () => {
-    return new Promise((resolve) => {
-      chrome.storage.local.getBytesInUse(null, (bytesInUse) => {
+const getUsedStorageSize = async () => {
+  return new Promise((resolve, reject) => {
+    chrome.storage.local.getBytesInUse(null, (bytesInUse) => {
+      if (chrome.runtime.lastError) {
+        reject(chrome.runtime.lastError);
+      } else {
         resolve(bytesInUse);
-      });
+      }
     });
-  }, []);
-
-  const removeAnyData = useCallback(async () => {
-    return new Promise((resolve) => {
-      chrome.storage.local.get(null, async (items) => {
-        const keys = Object.keys(items);
-        const notDeletedKeys = ["initialSearchValue"];
-        const deletableKeys = keys.filter(
-          (key) => !notDeletedKeys.includes(key)
-        );
-
-        if (deletableKeys.length === 0) {
-          return resolve();
-        }
+  });
+};
 
-        const deleteKey = deletableKeys[0];
-        await new Promise((res) => chrome.storage.local.remove(deleteKey, res));
+const calculateAvailableStorage = async () => {
+  const usedSize = await getUsedStorageSize();
+  return STORAGE_LIMIT - usedSize;
+};
 
-        resolve();
-      });
-    });
-  }, []);
+const calculateDataSize = (data) => {
+  const jsonString = JSON.stringify(data);
+  return new TextEncoder().encode(jsonString).length;
+};
 
-  const saveDataWithStorageCheck = useCallback(
-    async (key, data) => {
-      const newDataSize = new Blob([JSON.stringify(data)]).size;
-      let availableSpace = STORAGE_LIMIT - (await checkChromeStorageUsage());
+const deleteOldestData = async (requiredSpace) => {
+  const storedData = await new Promise((resolve) => {
+    chrome.storage.local.get(null, (items) => resolve(items));
+  });
 
-      while (availableSpace < newDataSize) {
-        await removeAnyData();
-        availableSpace = STORAGE_LIMIT - (await checkChromeStorageUsage());
-      }
+  const sortedKeys = Object.keys(storedData).sort((a, b) => {
+    return storedData[a].timestamp - storedData[b].timestamp;
+  });
+
+  let freeSpace = 0;
+  for (const key of sortedKeys) {
+    const dataSize = calculateDataSize(storedData[key]);
+    await new Promise((resolve) => chrome.storage.local.remove(key, resolve));
 
-      chrome.storage.local.set({ [key]: data });
-    },
-    [checkChromeStorageUsage, removeAnyData]
-  );
+    freeSpace += dataSize;
+    if (freeSpace >= requiredSpace) {
+      break;
+    }
+  }
+};
+
+const saveDataWithStorageCheck = async (data, keyword) => {
+  try {
+    const dataSize = calculateDataSize(data);
+    const availableSpace = await calculateAvailableStorage();
+
+    if (dataSize > availableSpace) {
+      await deleteOldestData(dataSize - availableSpace);
+    }
+
+    await new Promise((resolve) => {
+      chrome.storage.local.set({ [keyword]: data }, resolve);
```

Comment:
```suggestion
      chrome.storage.local.set({ [keyword]: data }, resolve);
```
이 부분 또한 Promise를 반환하는 것 같습니다. 다음은 [chrome Extension localStorage 메서드들의 공식문서](https://developer.chrome.com/docs/extensions/reference/api/storage?hl=ko)입니다.